src/main/resources/include/hail/NativeStatus.h

```diff
@@ -0,0 +1,46 @@
+#ifndef HAIL_NATIVESTATUS_H
+#define HAIL_NATIVESTATUS_H 1
+
+#include "hail/NativeObj.h"
+#include <string>
+#include <stdarg.h>
+
+namespace hail {
+
+class NativeStatus : public NativeObj {
+public:
+  int errno_;
+  std::string msg_;
+  std::string location_;
+  
+public:
+  inline NativeStatus() : errno_(0) { }
+  
+  virtual ~NativeStatus() { }
+  
+  inline void clear() {
+    // When errno_ == 0, the values of err_ and location_ are ignored
+    errno_ = 0;
+  }
+  
+  void set(const char* file, int line, int code, const char* msg, ...) {
+    char buf[8*1024];
+    sprintf(buf, "%s,%d", file, line);
```

Comment:
Should be in stdio, and I ought to have that in NativeStatus.h.  We probably get away
with it just because everything that #includes NativeStatus.h has an earlier <stdio.h>